### PR TITLE
Consolidate how Rails version is fetched for dummy-app and sandbox

### PR DESCRIPTION
**Pull request by waiting-for-dev**
## Summary

We use now the same that it's used for the sandbox.

## Checklist  

Check out our [PR guidelines](https://github.com/solidusio/.github/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [x] I have written a thorough PR description.
- [x] I have kept my commits small and atomic.
- [x] [I have used clear, explanatory commit messages](https://github.com/solidusio/.github/blob/main/CONTRIBUTING.md#writing-good-commit-messages).

The following are not always needed:

- 📖 I have updated the README to account for my changes.
- 📑 I have documented new code [with YARD](https://www.rubydoc.info/gems/yard/file/docs/Tags.md).
- 🛣️ I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).
- ✅ I have added automated tests to cover my changes.
- 📸 I have attached screenshots to demo visual changes.

### CODE REVIEW

File: bin/dummy-app [hunk 1]
```diff
@@ -14,7 +14,8 @@ function unbundled {
 test "$DB" = "sqlite" && export DB="sqlite3"
 
 rm -rf ./dummy-app
-rails ${RAILS_VERSION:+_${RAILS_VERSION}_} new dummy-app \
+rails_version=`bundle exec ruby -e'require "rails"; puts Rails.version'`
+rails _${rails_version}_ new dummy-app \
```
Comment (elia) [inline bin/dummy-app:18]: I've had issues with using this setup calling `bundle exec`, inside the script, thoughts on doing it the other way around and updating the sandbox instead?

Using the env var seems more adaptable to different situations and the Gemfile would be based on an environment variable anyway.

**Review by elia — CHANGES_REQUESTED** [review]

File: bin/dummy-app (hunk 1 above)
Comment (waiting-for-dev) [reply bin/dummy-app:18]: The issue is that in the script context, `$RAILS_VERSION` is expected to be an actual version, like `7.0` or `6.2`. However, in other places of the Solidus ecosystem is expected to be a semver reference, like `~> 7.0`. See [docker config on main Solidus](https://github.com/solidusio/solidus/blob/c09ead361b6b705897bf6a05aa38cef24d7d3c0d/docker-compose.yml#L36) or the [CircleCI orb](https://github.com/solidusio/circleci-orbs-extensions/blob/a86ae2bdf4b0b99466e5b36c96a5edcd3ad43b8d/src/commands/test-branch.yml#L28) as examples. (I use docker for development, and my docker-compose file had the env var; that was how it hit me :sweat_smile:).

Do you recall which problem you found with `bundle exec`? At first sight, it looks like the safest way to get the actual version that we're using.

Comment (elia) [reply bin/dummy-app:18]: No I think it was related to versions of bundler one nested into the other, but I don't have any reference, let's go with this code and if the problem comes up again we can switch to another solution 👍

**Review by elia — APPROVED** [review]
🙌

File: bin/dummy-app (hunk 1 above)
Comment (waiting-for-dev) [reply bin/dummy-app:18]: All right, thanks, @elia!

**Pull request merged**